c10/xpu/XPUFunctions.cpp

```diff
@@ -89,6 +89,10 @@ void initDeviceProperties(DeviceProp* device_prop, int device) {
 #define ASSIGN_DEVICE_ASPECT(member) \
   device_prop->has_##member = raw_device.has(sycl::aspect::member);
 
+#define ASSIGN_EXP_CL_ASPECT(member) \
+  device_prop->has_##member =        \
+      raw_device.ext_oneapi_supports_cl_extension("cl_intel_" #member);
```

Comment:
```suggestion
  sycl::ext::oneapi::experimental::cl_version version_##member;            \
  device_prop->has_##member = raw_device.ext_oneapi_supports_cl_extension( \
      "cl_intel_" #member, &version_##member);
```

The compilation does not fail, however I'm getting core dump without second argument:
```bash
$ python -c "import torch; print(torch.xpu.get_device_capability())"
[1]    130676 segmentation fault (core dumped)  python -c "import torch; print(torch.xpu.get_device_capability())"
```

Reply:
Thanks for your suggestion. I will check it.

Reply:
I'm using PTDB 0.5.2.18 with updated ubuntu 22.04 and python 3.9 venv installed from deadsnakes/ppa. Just reproduced the issue.

```
icpx --version
Intel(R) oneAPI DPC++/C++ Compiler 2024.1.3 (2024.1.3.20240604)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/intel/oneapi/compiler/2024.1/bin/compiler
Configuration file: /opt/intel/oneapi/compiler/2024.1/bin/compiler/../icpx.cfg
```

Reply:
OK, thanks. The compiler team has confirmed this is a bug.

Reply:
I will merge this PR once our CI upgrades to the appropriate compiler.

Reply:
Thank you!

Reply:
No problem!